### PR TITLE
Setup codeowners file with new team structure

### PR DESCRIPTION
**Pull request by 45930**
This PR cleans up the `CODEOWNERS` file and lays the groundwork for a new round-robin code review process that will give the exploration team and future zkapps team exposure to o1js PR reviews.

Ultimately, it is the `o1js-admins` job to manage what code gets into o1js.  This process aims to get more people to the level that they can be added to the list of admins.

### CODE REVIEW

File: CODEOWNERS
Comment (45930) [inline CODEOWNERS:9]: This is a legacy rule.  I'm happy to keep it, but I'm not sure if `crypto-eng-reviewers` is a valid target anymore.

Comment (Trivo25) [reply CODEOWNERS:9]: Let's remove this. `/provable` has a lot of stuff that won't require crypto engineer code reviews (eg the Merkle tree api and many other things)

File: CODEOWNERS
Comment (mrmr1993) [inline CODEOWNERS:4]: I think these rules operate as an 'or', not an 'and'. That is to say, any time someone from one team approves, the other team's review will no longer be required.

File: CODEOWNERS
Comment (45930) [reply CODEOWNERS:9]: Is there any specific part of `/provable` that you would want a crypto review on?

File: CODEOWNERS
Comment (45930) [inline CODEOWNERS:7]: This incorrect use of codeowners is still a more secure subset of our current rules AFAICT.  Right now any o1 eng can approve o1js PRs.  This will require a review from a subset of that group.

Reviews from crypto, protocol, or node specialists may also be required from time to time, which can be requested manually.  We probably also want to ensure that a github admin can override and merge an emergency PR if needed.

**Comment by 45930** [comment]
@mrmr1993 With the updated comments, does this procedure work for you?

File: CODEOWNERS
Comment (Trivo25) [inline CODEOWNERS:11]: I would be in favor of leaving these specifications out of the file until we see an actual need for it - right now it seems too arbitrary (other than the `examples` folder maybe). If we decide to make exceptions like these, we would have to add more than just this particular folder. On the other hands, it's hard to differentiate between changes that might seem trivial but are actually non-trivial

Comment (45930) [reply CODEOWNERS:11]: Ok, I will rm this for now.

Just so you know what I'm going for, codeowners can be used to require review of specific files by specific teams.  In this case (and thus reduce the load of review from other teams).  If a big PR includes API changes, new data types, tests, docs, and an example, maybe we only need admin review on the API changes and data types.  The other changes still need a review, but that review can come from one of the teams instead.

I will go with the simplest version to start, which is just `o1js-admin` is required by default and current and future zkapps teams will be required on changed to `/src/lib` since that's basically the current model.

**Review by Trivo25 — APPROVED** [review]
anything else we need to land this?

**Comment by 45930** [comment]
@Trivo25 I'll merge but we may still need admins to set up the round robin assignment.

**Pull request merged**
